spec/features/view_switcher_spec.rb

```diff
@@ -1,55 +1,40 @@
 require 'spec_helper'
 
 RSpec.feature 'View switcher' do
-  let(:current_user) do
-    double(
-      :webauth_user,
-      login: 'sunetid',
-      logged_in?: true,
-      permitted_apos: [],
-      is_admin: true,
-      roles: [],
-      groups: [],
-      permitted_collections: []
-    )
+  before :each do
+    admin_user # see spec_helper
   end
 
   feature 'is present' do
     scenario 'bulk update' do
-      expect_any_instance_of(ReportController).to receive(:current_user)
-        .at_least(1).times.and_return(current_user)
       visit report_bulk_url f: { objectType_ssim: ['item'] }
       within '.report-toggle' do
         expect(page).to have_css 'li.active a', text: 'Bulk Update View'
       end
     end
+
     scenario 'catalog results' do
-      expect_any_instance_of(CatalogController).to receive(:current_user)
-        .at_least(1).times.and_return(current_user)
       visit catalog_index_path f: { objectType_ssim: ['item'] }
       within '.report-toggle' do
         expect(page).to have_css 'ul.dropdown-menu-right li.active a', text: 'Results View'
       end
     end
+
     scenario 'report view' do
-      expect_any_instance_of(ReportController).to receive(:current_user)
-        .at_least(1).times.and_return(current_user)
       visit report_path f: { objectType_ssim: ['item'] }
       within '.report-toggle' do
         expect(page).to have_css 'li.active a', text: 'Report View'
       end
     end
+
     scenario 'discovery report' do
-      expect_any_instance_of(DiscoveryController).to receive(:current_user)
-        .at_least(1).times.and_return(current_user)
       visit discovery_url f: { objectType_ssim: ['item'] }
       within '.report-toggle' do
         expect(page).to have_css 'li.active a', text: 'Discovery Report View'
       end
     end
+
     scenario 'workflow grid' do
-      expect_any_instance_of(ReportController).to receive(:current_user)
-        .at_least(1).times.and_return(current_user)
       visit report_workflow_grid_url f: { objectType_ssim: ['item'] }
       within '.report-toggle' do
         expect(page).to have_css 'li.active a', text: 'Workflow Grid View'
```

Comment:
All these `current_user` specs appear to be spurious to the focus of this view spec.  All it requires is a vanilla admin user and it should not be putting any expectations on it.  These `expect_any_instance` are a reaction to potential problems with `allow_any_instance` and we don't really understand whether this `expect` pattern is better.  Given weird failures in the feature specs, I found it was better to remove all these and use just the admin_user from the spec helper for more consistent behavior.